### ISSUE TITLE
doc_markdown on lowerCamelCase.

### ISSUE THREAD

**Issue by TomFryersMidsummer** [rust-lang/rust-clippy#11568]
### Summary

Currently, doc_markdown flags `lowerCamelCase` identifiers, as well as `UpperCamelCase` ones. The former rarely occur as identifiers in Rust, so lots of these are likely to be false positives, such as `iOS` and `macOS`, which are in the default ignore list already, and `sRGB`, which I've just encountered. There's also (`rg "^[a-z][A-Za-z]*[A-Z][A-Za-z]*[A-Z][A-Za-z]?$" /usr/share/dict/words`) `fMRI`. And `eSATA`.

Maybe the lint should fire only if the first letter is upper case. Or perhaps extending the default ignore list would suffice.

### Lint Name

doc_markdown

### Reproducer

I tried this code:

```rust
/// Reads from sRGB.
fn from_srgb() {}
```

I saw this happen:

```
warning: item in documentation is missing backticks
 --> roof-scan/src/image.rs:1:16
  |
1 | /// Reads from sRGB.
  |                ^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
  = note: `-W clippy::doc-markdown` implied by `-W clippy::pedantic`
help: try
  |
1 | /// Reads from `sRGB`.
  |                ~~~~~~
```

I expected to see this happen: [no error].

### Version

```text
rustc 1.72.1 (d5c2e9c34 2023-09-13)
binary: rustc
commit-hash: d5c2e9c342b358556da91d61ed4133f6f50fc0c3
commit-date: 2023-09-13
host: x86_64-unknown-linux-gnu
release: 1.72.1
LLVM version: 16.0.5
```

**Comment by koka831** [comment]
Hi @TomFryersMidsummer 
I'm not sure if this is the solution you're looking for, but you can set the [`clippy.toml` to ignore it](https://rust-lang.github.io/rust-clippy/master/index.html#/doc_markdown) like this.

```toml
doc-valid-idents = ["sRGB", ".."]
```
